dfastmi/io/DFastMIConfigParser.py

```diff
@@ -0,0 +1,269 @@
+
+    def config_get(
+        self,
+        value_type: Type[T],
+        group: str,
+        key: str,
+        default: Optional[T] = None,
+        positive: bool = False
+    ) -> T:
+        """
+        Get a value from a selected group and keyword in the analysis settings.
+
+        Arguments
+        ---------
+        group : str
+            Name of the group from which to read.
+        key : str
+            Name of the keyword from which to read.
+        default : Optional[T]
+            Optional default value.
+
+        positive : bool
+            Flag specifying whether all values are accepted (if False), or only strictly positive values (if True).
+
+        Raises
+        ------
+        Exception
+            If the keyword isn't specified and no default value is given.
+            If a negative or zero value is specified when a positive value is required.
+
+        Returns
+        -------
+        val : T
+            Provided value type.
+        """
+        try:
+            if value_type is str:
+                entry_value = self._config[group][key]
+            else:
+                entry_value = self._config[group][key].lower()
+            val = self._config_processor.process_config_value(value_type, entry_value)[0]
+        except:
+            if not default is None:
+                val = default
+            else:
+                raise Exception(f'No {value_type.__name__} value specified for required keyword "{key}" in block "{group}".')
+
+        if positive:
+            if val <= 0:
+                raise Exception(f'Value for "{key}" in block "{group}" must be positive, not {val}.')
+        return val
+
+    def config_get_range(
+        self,
+        group: str,
+        key: str,
+        default: Optional[Tuple[float,float]] = None,
+    ) -> Tuple[float, float]:
+        """
+        Get a start and end value from a selected group and keyword in the analysis settings.
+
+        Arguments
+        ---------
+        group : str
+            Name of the group from which to read.
+        key : str
+            Name of the keyword from which to read.
+        default : Optional[Tuple[float,float]]
+            Optional default range.
+
+        Returns
+        -------
+        val : Tuple[float,float]
+            Lower and upper limit of the range.
+        """
+        try:
+            ini_value = self.config_get(str, group, key, "")
+            obrack = ini_value.find("[")
+            cbrack = ini_value.find("]")
+            if obrack >= 0 and cbrack >= 0:
+                ini_value = ini_value[obrack + 1 : cbrack - 1]
+            vallist = [float(fstr) for fstr in ini_value.split(":")]
+            if vallist[0] > vallist[1]:
+                val = (vallist[1], vallist[0])
+            else:
+                val = (vallist[0], vallist[1])
+        except:
+            if not default is None:
```

Comment:
if not default is None: --> if default is not None

Reply:
Ok, changed